app/controllers/api/v1/apps_controller.rb

```diff
@@ -20,10 +20,12 @@ def application_options
   end
 
   def app_scopes_or_default
-    app_params[:scopes] || Doorkeeper.configuration.default_scopes
+    Array(app_params[:scopes]).first || Doorkeeper.configuration.default_scopes
```

Comment:
There's a note in the spec about this which links to https://github.com/mastodon/mastodon/issues/30152 which sort of explains the `scope`/`scopes` flat/nested dynamic.

The change here preserves the existing behavior from the scope. If that issue is resolved some other way, we'd want to update accordingly.

Reply:
How does this change fix #30152? I'm a bit confused by this code?

Reply:
I don't think it fixes anything (unless as an accidental side effect it does) ... I think it was an effort to preserve the existing behavior, while allowing either the flat single param form or the array form, but keep the specs passing.

Now that I look at it though, the spec for the array case is a little ambiguous in that the assertion around the selected scope (read) is the same value of "the first item from this array" and "the doorkeeper default". I think it may be actually asserting the latter, but my change here is basically doing the former.

Reply:
yeah, it should be "if scopes are passed, either as a string or an array, then they should be used" — if we accept an array of scopes, and don't return a 422, then they should be respected, currently they are silently ignore and the default scope is used, despite the user "passing scopes"

Reply:
Ah, in that case then I think this change is in fact fixing things in that direction (allowing arrays), but then also breaking things in another direction (only grabbing the FIRST value from a provided array) because I misunderstood the spec. Will update this and/or do separate PR for array support here, attempting to make it less likely to pass spec for wrong reasons.

Reply:
Technically the spec says `scope` should be a space separated scopes, but we use `scopes` so could support either.

I'd be okay with rejecting arrays of scopes, or with accepting them, but whatever the behaviour it should be consistent with our other APIs. But to silently accept but not use is definitely a bad/broken API behaviour.